core/arch/arm/mm/mobj_dyn_shm.c

```diff
@@ -120,21 +160,38 @@ static void *mobj_reg_shm_get_va(struct mobj *mobj, size_t offst, size_t len)
 
 static void reg_shm_unmap_helper(struct mobj_reg_shm *r)
 {
+	size_t burst_pages = r->mm->size;
+	size_t burst_count = 1;
+	vaddr_t va = 0;
+	size_t i = 0;
+
 	assert(r->mm);
 	assert(r->mm->pool->shift == SMALL_PAGE_SHIFT);
-	core_mmu_unmap_pages(tee_mm_get_smem(r->mm), r->mm->size);
+
+	if (IS_ENABLED(CFG_DYN_SHM_INTERRUPTIBLE_LOCK)) {
+		/*
+		 * Unmap pages one by one to prevent masking interrupts
+		 * over a potentially long period of time.
+		 */
+		burst_pages = 1;
+		burst_count = r->mm->size;
+	}
+
+	va = tee_mm_get_smem(r->mm);
+	for (i = 0; i < burst_count; i++)
+		core_mmu_unmap_pages(va + i * SMALL_PAGE_SIZE, burst_pages);
```

Comment:
How about releasing the lock in `core_mmu_unmap_pages()` just after the lookup with `find_map_by_va()` instead? That way we don't need to care about `burst_pages`.